main.tf

```diff
@@ -22,14 +21,14 @@ resource "aws_ecr_lifecycle_policy" "lifecycle_policy" {
 }
 
 resource "null_resource" "replicate_lifecycle_policy" {
-  count = length(var.lifecycle_policy) > 0 ? length(var.replicated_region) : 0
+  for_each = length(var.lifecycle_policy) > 0 ? toset(var.replicated_region) : toset([])
```

Comment:
We can either use `toset(var.replicated_region)` which could cause an issue during a plan if the variable contains values that can only be known at apply

the other solution is to edit the trigger to use the value of the `for_each/count`, which would have the "issue" of triggering for all elements if we were to add an element in any other places than the end of the list

```
triggers = {
  policy_hash = sha256(var.lifecycle_policy)
  region = each.value # or var.replicated_region[count.index]
}
```